File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WinPropertyString.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_WinPropertyString
#/
class t_WinPropertyString:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_string = ""	#std::string
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_string = ""

	def read(self,cVariable):
		self.m_string = cVariable.getString(65535)
		return True

	def write(self,cVariable):
		cVariable.putString(self.m_string,65535)
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WinPropertyString.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WinPropertyString
*/
public class t_WinPropertyString : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public string	m_string;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WinPropertyString() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_string = "";
	}
	public bool read(CReadVariable cVariable) {
		try {
			cVariable.getString(ref m_string,65535);
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(ref m_string,65535);
		return true;
	}
};

